livebooks/planned-disruptions.livemd

```diff
@@ -0,0 +1,174 @@
+# Planned Disruptions
+
+## Section
+
+```elixir
+defmodule Dotcom.Utils.Datetime do
+  def in_range?({start, nil}, datetime) do
+    Timex.after?(datetime, start)
+  end
+
+  def in_range?({start, stop}, datetime) do
+    Timex.after?(datetime, start) && Timex.before?(datetime, stop)
+  end
+
+  def later() do
+    {_, end_of_next_week} = next_week()
+    beginning_of_later = Timex.shift(end_of_next_week, seconds: 1)
+
+    {beginning_of_later, nil}
+  end
+
+  def later?(datetime) do
+    later()
+    |> in_range?(datetime)
+  end
+
+  def next_week() do
+    now = Util.now()
+    end_of_week = Timex.end_of_week(now) |> Util.end_of_service()
+    beginning_of_next_week = Timex.shift(end_of_week, seconds: 1)
+    end_of_next_week = beginning_of_next_week |> Timex.end_of_week() |> Util.end_of_service()
+
+    {beginning_of_next_week, end_of_next_week}
+  end
+
+  def next_week?(datetime) do
+    next_week()
+    |> in_range?(datetime)
+  end
+
+  def overlap?(range, {start, stop}) do
+    in_range?(range, start) || in_range?(range, stop)
+  end
+
+  def this_week() do
+    now = Util.now()
+    end_of_week = now |> Timex.end_of_week() |> Util.end_of_service()
+
+    {now, end_of_week}
+  end
+
+  def this_week?(datetime) do
+    this_week()
+    |> in_range?(datetime)
+  end
+
+  def today() do
+    now = Util.now()
+    end_of_day = now |> Timex.end_of_day() |> Util.end_of_service()
+
+    {now, end_of_day}
+  end
```

Comment:
Have we confirmed we want "today" to start from **now**, and not from the beginning of today?

Reply:
Yeah, I changed that in the actual code.